### PR TITLE
feat(frontend): show usd value in token selector

### PR DESCRIPTION
**Pull request by tommy0x**
### Changes Made

- Extend `useMulticallTokenBalance` hook to also return usd values
- Pass usd values down to CoinSelector

### Additional Notes

⚠️ This PR uses #322 

### Screenshots/videos

Screenshot 2023-10-13 at 11 36 05

### Checklist

Please tick the boxes that apply and provide additional details where necessary. Add or edit items as needed.

- [x] Changes are limited to a single goal.
- [x] Responsive design has been tested and looks good on all devices and screen sizes.
- [x] Changes have been tested on all major browsers (Chrome, Firefox, Safari).
- [x] The changes in Chromatic UI Tests all look good.
- [x] No accessibility issues have been introduced in Storybook's Accessibility tab.
- [x] The code has been optimized for performance.

### CODE REVIEW

File: packages/frontend/src/hooks/useMulticallTokenBalance.tsx [hunk 1]
```diff
@@ -1,86 +1,107 @@
...
-            balanceMap.set(tokenAddress, formattedTokenBalance);
-            continue;
-          }
-
-          balanceMap.set(tokenAddress, '0');
-                  
-        } catch (error) {
-          console.error(error);
+    const usdPricesForTokens = await Promise.allSettled(
+      tokens.map(token => sifi.getUsdPrice(chainId, token.address))
```
Comment (immutablegoblin) [inline packages/frontend/src/hooks/useMulticallTokenBalance.tsx:43]: Could we do this after we have fetched multicall token balances, so we only call this endpoint for tokens with wallet balance? Would save a lot of API calls.

Comment (immutablegoblin) [reply packages/frontend/src/hooks/useMulticallTokenBalance.tsx:43]: Alternatively, we could do it the other way round, where we do the multicall after fetching the USD balances?
We are still getting easily rate limited on that so worth experimenting with.

**Review by immutablegoblin — DISMISSED** [review]

**Comment by immutablegoblin** [comment]
Finding this hard to actually test - but it's due to the poor balance multicall, not this PR.
Will be looking into that shortly, it's barely functional at the moment.

**Review by boriskubrik — APPROVED** [review]
Let's ship it and look into a better solution in the future.

**Pull request merged**